### PR TITLE
docs(functional): cover sequence

### PR DESCRIPTION
**Pull request by probeiuscorp**
## Related Bug Reports or Discussions

Fixes #2724

## Summary

Show some examples of `sequence` to expand on advantages of seeing atoms as monads. Also add a list of other good resources to look at for monads.

## Check List

- [x] `pnpm run prettier` for formatting code and docs

### CODE REVIEW

File: docs/basics/functional-programming-and-jotai.mdx [hunk 1]
```diff
@@ -38,31 +38,97 @@ const greetingPromise = (async function () {
...
+The shared heritage of Promises and Atoms means many patterns and best-practices
+can be reused between them. Let's take a look at one.
+
+### Sequencing
+
+When talking about callback hell, we often mention the boilerplate, the
+indentation and the easy-to-miss mistakes. However, plumbing a single async
+operation into another single async operation was not the end of the callback
+struggle. What if we made four network calls and needed to wait for them all?
+A snippet like this was common:
```
Comment (probeiuscorp) [inline docs/basics/functional-programming-and-jotai.mdx:76]: Am I correct here? I did not code collaboratively before the ES6 Promise, so this is just from my experience.

File: docs/basics/functional-programming-and-jotai.mdx [hunk 2]
```diff
@@ -38,31 +38,97 @@ const greetingPromise = (async function () {
 })()
 ```
 
-This similarity is no coincidence. Both atoms and promises are **Monads**, a
-concept from **functional programming**. The syntax used in both examples is
-called **do-notation**, a syntax sugar for the plainer monad interface.
+This similarity is no coincidence. Both atoms and promises are **Monads**†, a
```
Comment (dmaskasky) [inline docs/basics/functional-programming-and-jotai.mdx:41]:
```suggestion
This similarity is no coincidence. Both atoms and promises are **Monads**, a
```

File: docs/basics/functional-programming-and-jotai.mdx (hunk 1 above)
Comment (dai-shi) [reply docs/basics/functional-programming-and-jotai.mdx:76]: I'm not 100% sure if it's correct...

**Review by dai-shi — APPROVED** [review]
let's merge anyway

**Pull request merged**